modules/mod_acl_user_groups/support/acl_user_groups_checks.erl

```diff
@@ -587,8 +587,23 @@ can_rsc_for_collab(Id, Action, CGId, CatId, UGs, Context) ->
     andalso (
         can_rsc_for_collab_ugs(Id, Action, CGId, CatId, UGs, Context)
         orelse can_rsc_collab_group_content(Action, CGId, UGs, Context)
+        orelse has_unrestricted_collab_content_user_groups(Context)
     ).
 
+has_unrestricted_collab_content_user_groups(Context) ->
+    AllUGs = sets:from_list(user_groups_all(Context)),
+    UnrestrictedUGs = sets:from_list(
+        lists:map(fun (UG) -> m_rsc:rid(UG, Context) end,
+        unrestricted_collab_content_user_groups())
+    ),
+    not sets:is_disjoint(AllUGs, UnrestrictedUGs).
+
+unrestricted_collab_content_user_groups() ->
```

Comment:
Maybe add a TODO here?

I could also imagine that we could make those user groups or users "unrestricted" if they have permission to delete or insert random collaboration groups.

So in this case we could check if the user has permission to insert a (not insert-own, but really insert) a collaboration group into the Content-Group of the Collaboration-Group we are checking.